modules/browsemap/src/main/java/com/sygic/modules/browsemap/viewmodel/BrowseMapFragmentViewModel.kt

```diff
@@ -18,17 +33,71 @@ class BrowseMapFragmentViewModel private constructor(attributesTypedArray: Typed
         attributesTypedArray?.let {
             compassEnabled.value = it.getBoolean(R.styleable.BrowseMapFragment_sygic_compassEnabled, false)
             compassHideIfNorthUp.value = it.getBoolean(R.styleable.BrowseMapFragment_sygic_compassHideIfNorthUp, false)
+            mapInteractionMode.value =
+                    it.getInt(R.styleable.BrowseMapFragment_sygic_mapInteractionMode, MapInteractionMode.NONE)
             positionLockFabEnabled.value =
                     it.getBoolean(R.styleable.BrowseMapFragment_sygic_positionLockFabEnabled, false)
             zoomControlsEnabled.value = it.getBoolean(R.styleable.BrowseMapFragment_sygic_zoomControlsEnabled, false)
             it.recycle()
         }
+
+        mapInteractionMode.value?.let { mode ->
+            when (mode) {
+                MapInteractionMode.FULL, MapInteractionMode.MARKERS_ONLY -> mapInteractionManager.addOnMapClickListener(
+                    this
+                )
+            }
+        }
+    }
+
+    override fun onMapObjectsRequestStarted() {
+        extendedMapDataModel.removeOnClickMapMarker()
+    }
+
+    override fun onMapObjectsReceived(viewObjects: List<ViewObject>) {
+        mapInteractionMode.value?.let { mode ->
+            val firstViewObject = viewObjects.first()
```

Comment:
to first je asi docasne, ze? Zatail sa spoliehame, ze ak je tam marker, je vzdy prvy?

Reply:
Mno, za mě to bylo takto zatím final :) Nevím, marker byl vždy první a když nebyl tak mě zajímal také vždy první. Mám tedy tu logiku z navi přenést i sem?

Reply:
zatial to nechaj, ale viem ze v navi sa to menilo, lebo tam boli uiobjecty schovane za merker a neviem co vsetko